### PR TITLE
DROOLS-2578 [DMN Designer] Add use of 'DMNModelInstrumentedBase.setParent(..)'...

### PR DESCRIPTION
**Pull request by tarilabs**
...to marshaller

[DMN Designer] Add use of 'DMNModelInstrumentedBase.setParent(..)' to
marshaller

/cc @etirelli @manstis @jomarko

### CODE REVIEW

**Comment by tarilabs** [comment]
```
Test Result (1 failure / +1)
org.kie.workbench.common.services.backend.compiler.DefaultMavenCompilerTest.buildWithCloneTest
```

Jenkins test again

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DefinitionsConverter.java [hunk 1]
```diff
@@ -43,12 +43,19 @@ public static Definitions wbFromDMN(final org.kie.dmn.model.api.Definitions dmn)
         result.getNsContext().putAll(dmn.getNsContext());
 
         for (org.kie.dmn.model.api.ItemDefinition itemDef : dmn.getItemDefinition()) {
-            ItemDefinition itemDefConvered = ItemDefinitionPropertyConverter.wbFromDMN(itemDef);
-            result.getItemDefinition().add(itemDefConvered);
+            ItemDefinition itemDefConverted = ItemDefinitionPropertyConverter.wbFromDMN(itemDef);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DefinitionsConverter.java]: Should we cover ```drgElement```, ```artifact```, ```elementCollection``` and ```businessContextElement``` too?

Comment (tarilabs) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DefinitionsConverter.java]: Concerning `drgElement`, `artifact`, it was previously requested to manage them separately, in the Node of the graph, which is done at a DMNMarshaller level.
Concerning `elementCollection` this is basically to depict DRDs but it is not clear to me how should be managed from the stunner pov.
Concerning `businessContextElement` this is another non-execution element, but if it's covered by the editor is just matter to understand how to most appropriately convert from the stunner PoV (so as a graph node, or as a child element here?)

Nevertheless I would suggest all the above is a different scope than the current, which is covering setParent for existing converter as per DROOLS-2578, and I would suggest to manage as a separate JIRA/+PR once the Stunner requirements is identified?

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DefinitionsConverter.java]: Thanks for the explanations @tarilabs 

I agree, we can worry about the other items (```elementCollection``` and ```businessContextElement```) when the UI handles them and we understand what our (UI) requirements are better.

**Comment by tarilabs** [comment]
Hi, just a follow-up --can this be merged given it's approved, please?

**Comment by manstis** [comment]
@tarilabs Waiting on @jomarko to approve before merge.

A lot of the "workbench" related repositories need PRs to be both _peer_ and _QE_ reviewed before merge.

**Comment by tarilabs** [comment]
Any estimation on this, please? Just trying to avoid having the need of rebase due to potential conflicts. Thanks!

**Comment by tarilabs** [comment]
@jomarko 

> unguearded occurence in DMNMarshaller.connectRootWithChild is it fine?

I would say Yes because connectRootWithChild is never called with either of the params as null since original definition of the method

**Comment by tarilabs** [comment]
Can I have the error message please?

**Comment by jomarko** [comment]
@tarilabs sure, sorry I forgot to attach. However not saying much.
- Ui error
```
Uncaught exception: (TypeError) : Cannot read property 'Lyb' of null
```
- Server log error, however present also on master
```
08:39:06,931 ERROR [org.kie.dmn.core.compiler.DMNEvaluatorCompiler] (EJB default - 2) No expression defined for name 'Decision-1' on node 'Decision-1'
```

**Comment by tarilabs** [comment]
> `08:39:06,931 ERROR [org.kie.dmn.core.compiler.DMNEvaluatorCompiler] (EJB default - 2) No expression defined for name 'Decision-1' on node 'Decision-1'`

I cannot see how this is related with this change.

Moreover the reported error looks like the workbench is trying to compile the DMN model, and if a Decision is missing an Expression (which allegedly given the reported step was indeed not defined) it is normaly that DMN model compilation will fail.
Is the Workbench now trying to compile the DMN model when is being opened?
How this is impacted by this serialization work, please?
@manstis can you kindly advise?

**Comment by manstis** [comment]
@jomarko Hi, was the error in ```kie-(drools)-wb``` or something else please?

@tarilabs The ```kie-(drools)-wb``` **will** add the DMN file to ```KieBuilder``` and perform an incremental compilation of the project to get the list of errors (in the project) to show to Users (like your IDE shows a "problems" panel so does the Workbench).

I'll take a look at this PR this morning.

**Comment by jomarko** [comment]
@manstis as usually, I used `kie-wb-distributions/kie-drools-wb-parent` module.

**Comment by manstis** [comment]
@tarilabs @jomarko The error is caused by marshalling of the _enriched_ (parent property) model from the server to the client. I'll need to dig a bit more as to what is causing it and why; but it's something relating to a ```null``` value somewhere....

**Pull request merged**